pkg/handlers/lifecycle/cni/multus/template.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package multus
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// templateValuesFunc returns a template function that parses the Multus values template
+// and replaces socket-related placeholders with the provided readiness socket path.
+func templateValuesFunc(readinessSocketPath string) func(*clusterv1.Cluster, string) (string, error) {
+	return func(_ *clusterv1.Cluster, valuesTemplate string) (string, error) {
+		t, err := template.New("").Parse(valuesTemplate)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse Multus values template: %w", err)
+		}
+
+		type input struct {
+			ReadinessSocketPath string
+			SocketVolumeName    string
+		}
+
+		// Extract volume name from socket path
+		// e.g., "/run/cilium/cilium.sock" -> "cilium-sock"
+		// or "/var/run/calico/cni-server.sock" -> "calico-sock"
+		socketVolumeName := extractVolumeName(readinessSocketPath)
+
+		templateInput := input{
+			ReadinessSocketPath: readinessSocketPath,
```

Comment:
We should be able to simplify this and only pass the socket `readinessSocketPath`, leave everything else to the yaml template.